h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -1369,7 +1374,7 @@ else if (_parms._weights_column != null && _weights != null && !_weights.isBinar
       if (isResponseOptional() && _parms._response_column != null && _response == null) {
         _vresponse = va.vec(_parms._response_column);
       }
-      _valid = adaptFrameToTrain(va, "Validation Frame", "_validation_frame", expensive);
+      _valid = adaptFrameToTrain(va, "Validation Frame", "_validation_frame", expensive, false); //FIXME: this encodes validation frame BEFORE TE is applied
```

Comment:
creating a ticket for this one: this is closely related  to PUBDEV-7775 but not exactly equivalent.
1. we can wonder why in some circumstances only known to `adaptFrameToTrain` and not made explicit at the time this method is called, this method will decide to encode the categorical columns... which makes it impossible to work in combination with an additional encoder.
2. I have an experiment in local branch that modifies `adaptFrameToTrain` so that the encoder is injected as a param instead of being guessed. This allows to inject an encoder that does both TE + categorical encoding.
3. This is an issue only when model `categorical_encoding != AUTO`
4. AutoML always uses `categorical_encoding == AUTO` for now.
5. AutoML is the only "client" that can use this TE integration.
6. Therefore, I suggest to fix this concern in a separate PR.

Reply:
created follow-up tickt PUBDEV-7785 instead.